### PR TITLE
device.coffee: Remove hardcode of resin-boot partitions's mount point

### PR DESCRIPTION
**Pull request by telphan**
@pcarranzav @agherzan @floion *@emirotin (Sensei)

This PR should go hand in had with this: https://github.com/resin-os/meta-resin/pull/251

Signed-off-by: Theodor Gherzan <theodor@gherzan.ro>

---
This change is [Reviewable](https://reviewable.io/reviews/resin-io/resin-supervisor/245)

### CODE REVIEW

**Comment by pcarranzav** [comment]
@resin-jenkins add to whitelist

File: src/device.coffee [hunk 1]
```diff
@@ -44,10 +44,19 @@ exports.getID = do ->
 exports.reboot = ->
 	request.postAsync(config.gosuperAddress + '/v1/reboot')
 
+exec = require('child_process')
```
Comment (emirotin) [inline src/device.coffee]: this should go to the top section, and better be `{ execSync } = require('child_process ')` (then just use this bare function)

**Comment by pcarranzav** [comment]
@telphan I see a few issues with this:
* This code will throw synchronously if it fails, crashing the supervisor
* The `exec` definition would better go above with the other requires (and can be combined with the `execAsync` require).
* It would be better to make this function asynchronous, using `execAsync`, returning a Promise and changing `setBootConfig` so that it fulfills the promise to get the mountpoint.
* I don't know if /proc/mounts will return the proper mountpoint inside the container, and if so, it probably still needs `/mnt/root/` to be prepended.

Welcome to coffeescript ;)

All in all, this approach will work if we do those changes, but I wonder if it would be better to pass the mount point as an environment variable to the supervisor (as we do with config.json), defaulting to the old value for backwards compatibility. In that case, we just add an entry for this in `config.coffee` and then use `config.bootMountPoint` here. Maybe @Page- or @petrosagg can weigh in.

File: src/device.coffee [hunk 2]
```diff
@@ -44,10 +44,19 @@ exports.getID = do ->
...
+
+findMountPointByPartLabel = (label) ->
+  command = "grep $(blkid -L #{label}) /proc/mounts"
+  try
+    return (exec.execSync(command).toString().split " ")[1]
```
Comment (Page-) [inline src/device.coffee]: This should be `return exec.execSync(command).toString().split(' ')[1]`

File: src/device.coffee [hunk 3]
```diff
@@ -44,10 +44,19 @@ exports.getID = do ->
...
+  catch err
+    throw new Error ("Could not determine mount point for partition #{label}: #{stderr}, ExitCode: #{error.code}", { error: body.Error }, 'Set log to display error')
```
Comment (Page-) [inline src/device.coffee]: The `Error` constructor only takes one argument, I'm not sure what the second two actually are either?

File: src/device.coffee [hunk 4]
```diff
@@ -47,7 +47,7 @@ exports.reboot = ->
 exports.hostConfigEnvVarPrefix = 'RESIN_HOST_'
 bootConfigEnvVarPrefix = 'RESIN_HOST_CONFIG_'
 bootBlockDevice = '/dev/mmcblk0p1'
-bootMountPoint = '/mnt/root/boot'
+bootMountPoint = '/mnt/root/' + config.bootMountPoint
```
Comment (pcarranzav) [inline src/device.coffee]: Looks like the last `/` in `/mnt/root/` should be removed here

File: src/config.coffee [hunk 5]
```diff
@@ -49,3 +49,4 @@ module.exports =
 		'RESIN_API_KEY'
 	]
 	dataPath: checkString(process.env.RESIN_DATA_PATH) ? '/resin-data'
+  bootMountPoint: checkString(process.env.BOOT_MOUNTPOINT) ? '/boot'
```
Comment (pcarranzav) [inline src/config.coffee]: Tabs, not spaces ;)

Comment (telphan) [inline src/config.coffee]: Fixing.

File: src/device.coffee (hunk 4 above)
Comment (telphan) [inline src/device.coffee]: You would be correct.

**Comment by pcarranzav** [comment]
:lgtm:

**Pull request merged**